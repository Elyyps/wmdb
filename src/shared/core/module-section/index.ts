export * from "./module-section.component";
