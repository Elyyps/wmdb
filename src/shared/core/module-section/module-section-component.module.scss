.module-section {
  position: relative;

  &__pattern {
    z-index: 0;
    position: absolute;
    left: 0;
    right: 0;
    top: 0;
    bottom: 0;

    &:after {
      content: "";
      background: linear-gradient(to bottom, rgba(255, 255, 255, 0.8), rgba(255, 255, 255, 0.7));
      position: absolute;
      left: 0;
      right: 0;
      top: 0;
      bottom: 0;
    }
  }
  &__background {
    z-index: 0;
    position: absolute;
    left: 0;
    right: 0;
    top: 0;
    bottom: 0;
    overflow: hidden;

    img {
      width: 100%;
      height: 100%;
      object-fit: cover;
      filter: blur(22px);
    }
  }

  &__foreground {
    position: relative;
    z-index: 0;
  }

  &-padding {
    &-top--120px {
      padding-top: 120px;
      &-mobile {
        @media (max-width: $breakpoint-small-max) {
          padding-top: 120px !important;
        }
      }
    }
    &-top--72px {
      padding-top: 72px;
      &-mobile {
        @media (max-width: $breakpoint-small-max) {
          padding-top: 72px !important;
        }
      }
    }
    &-top--40px {
      padding-top: 40px;
      &-mobile {
        @media (max-width: $breakpoint-small-max) {
          padding-top: 40px !important;
        }
      }
    }
    &-top--32px {
      padding-top: 32px;
      &-mobile {
        @media (max-width: $breakpoint-small-max) {
          padding-top: 32px !important;
        }
      }
    }
    &-top--0px {
      padding-top: 0px;
      &-mobile {
        @media (max-width: $breakpoint-small-max) {
          padding-top: 0px !important;
        }
      }
    }
    &-bottom--120px {
      padding-bottom: 120px;
      &-mobile {
        @media (max-width: $breakpoint-small-max) {
          padding-bottom: 120px !important;
        }
      }
    }
    &-bottom--72px {
      padding-bottom: 72px;
      &-mobile {
        @media (max-width: $breakpoint-small-max) {
          padding-bottom: 72px !important;
        }
      }
    }
    &-bottom--40px {
      padding-bottom: 40px;
      &-mobile {
        @media (max-width: $breakpoint-small-max) {
          padding-bottom: 40px !important;
        }
      }
    }
    &-bottom--32px {
      padding-bottom: 32px;
      &-mobile {
        @media (max-width: $breakpoint-small-max) {
          padding-bottom: 32px !important;
        }
      }
    }
    &-bottom--0px {
      padding-bottom: 0px;
      &-mobile {
        @media (max-width: $breakpoint-small-max) {
          padding-bottom: 0px !important;
        }
      }
    }
  }
  @media (max-width: $breakpoint-small-max) {
    padding-top: 32px;
    padding-bottom: 32px;
  }

  &__title {
    margin: 0 0 56px;
    padding: 0 0 26px;
    position: relative;
    text-align: center;
    @media (max-width: $breakpoint-small-max) {
      margin: 0 0 32px;
      padding: 0 0 20px;
    }
    &::after {
      position: absolute;
      left: 50%;
      bottom: 0;
      content: "";
      width: 150px;
      height: 2px;
      background: currentColor;
      transform: translateX(-50%);
    }
  }

  &__columns {
    @media (max-width: $breakpoint-xsmall-max) {
      > div {
        margin-bottom: 16px;
      }
    }
  }

  &--center {
    text-align: center;
  }

  &--small {
    margin: 0 auto;
    max-width: $breakpoint-small-medium;
  }
}
