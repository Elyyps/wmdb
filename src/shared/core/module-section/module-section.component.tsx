import * as React from "react";
import styles from "./module-section-component.module.scss";
import classNames from "classnames";

export interface IModuleSectionComponentProps {
  align?: string;
  backgroundColor?: string;
  children?: any;
  paddingBottom?: string;
  paddingTop?: string;
  pattern?: boolean;
  size?: string;
  title?: string;
}

const ModuleSectionComponent = (props: IModuleSectionComponentProps) => {
  const sectionClassName = classNames("section", {
    [`section--pattern`]: props.pattern,
    [`section--${props.align}`]: props.align,
    [`section--${props.size}`]: props.size
  });

  const style = {
    backgroundColor: props.backgroundColor,
    paddingBottom: props.paddingBottom,
    paddingTop: props.paddingTop
  };

  return (
    <div className={sectionClassName} style={style}>
      <div className="uk-container">
        {props.title && <h2 className={styles["module-section__title"]}>{props.title}</h2>}
        <div>{props.children}</div>
      </div>
    </div>
  );
};

export { ModuleSectionComponent };
