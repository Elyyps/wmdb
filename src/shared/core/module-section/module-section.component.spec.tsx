import React from "react";
import { shallow } from "enzyme";
import { ModuleSectionComponent } from "./module-section.component";

describe("[ModuleSection]", () => {
  it("should render component without crashing", () => {
    const renderedComponent = shallow(<ModuleSectionComponent />);
    expect(renderedComponent).toMatchSnapshot();
  });
});
