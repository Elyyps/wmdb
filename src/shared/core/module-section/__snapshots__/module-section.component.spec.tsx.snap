// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[ModuleSection] should render component without crashing 1`] = `
<div
  className="section"
  style={
    Object {
      "backgroundColor": undefined,
      "paddingBottom": undefined,
      "paddingTop": undefined,
    }
  }
>
  <div
    className="uk-container"
  >
    <div />
  </div>
</div>
`;
